tuf/api/metadata.py

```diff
@@ -300,14 +300,22 @@ class Signed:
         spec_version: The TUF specification version number (semver) the
             metadata format adheres to.
         expires: The metadata expiration datetime object.
+        unrecognized_fields: An optional dictionary storing all unrecognized
+            fields. Used for backward compatibility. If None is provided, an
+            empty dictionary will be created.
```

Comment:
```suggestion
        unrecognized_fields: Dictionary of all unrecognized fields.
```
The attribute is not optional I think (the constructor argument is). Also this isn't specifically for backwards (forwards?) compat: IIRC Spec requires this.